File: 01_GettingStarted/library/WebScoket/SocketClient.js
const socket = new WebSocket('ws://localhost:8080');

socket.onopen = function(event) {
  console.log('WebSocket connection established');
};

socket.onerror = function(error) {
  console.error('WebSocket error: ' + error);
};

socket.onmessage = function(event) {
  console.log('Message received from server: ' + event.data);
};

socket.onclose = function(event) {
  console.log('WebSocket connection closed');
};

File: 01_GettingStarted/library/WebScoket/SocketClient.go
package main

import (
	"fmt"
	"net"
	"os"
)

func main() {
	// Connect to the remote TCP server
	conn, err := net.Dial("tcp", "127.0.0.1:8080")
	if err != nil {
		fmt.Println("Error connecting:", err.Error())
		os.Exit(1)
	}

	// Send data to the server
	message := "Hello, server"
	_, err = conn.Write([]byte(message))
	if err != nil {
		fmt.Println("Error sending data:", err.Error())
		os.Exit(1)
	}

	// Receive data from the server
	buf := make([]byte, 1024)
	n, err := conn.Read(buf)
	if err != nil {
		fmt.Println("Error receiving data:", err.Error())
		os.Exit(1)
	}

	// Print the received data
	fmt.Println("Received data:", string(buf[:n]))

	// Close the connection
	conn.Close()
}
